tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * ownCloud - mediametadata
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Jalpreet Singh Nanda (:imjalpreet) <jalpreetnanda@gmail.com>
+ * @copyright Jalpreet Singh Nanda (:imjalpreet) 2016
+ */
+
+namespace OCA\MediaMetadata\Hooks;
+
+
+use OCA\MediaMetadata\AppInfo\Application;
+use OCA\MediaMetadata\Tests\TestCase;
+
+/**
+ * Class ImageHooksTest
+ *
+ * @package OCA\MediaMetadata\Hooks
+ */
+class ImageHooksTest extends TestCase {
+	/**
+	 * @var \OCA\MediaMetadata\Hooks\ImageHooks
+	 */
+	protected $imageHooks;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OC\Files\Node\Root
+	 */
+	protected $root;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ImageDimensionMapper
+	 */
+	protected $mapper;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCP\IDb
+	 */
+	protected $db;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ExtractMetadata
+	 */
+	protected $extractor;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\StoreMetadata
+	 */
+	protected $dbManager;
+	/**
+	 * @var $container
+	 */
+	private $container;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$app = new Application();
+		$this->container = $app->getContainer();
+
+		$this->root = $this->getMockBuilder('OC\Files\Node\Root')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->db = $this->getMockBuilder(
+			'\OCP\IDBConnection')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->mapper = $this->getMockBuilder('OCA\MediaMetadata\Services\ImageDimensionMapper')
+			->setConstructorArgs([
+				$this->db,
+			])
+			->getMock();
+		$this->extractor = $this->getMockBuilder(
+			'OCA\MediaMetadata\Services\ExtractMetadata')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->dbManager = $this->getMockBuilder(
+			'\OCA\MediaMetadata\Services\StoreMetadata')
+			->setConstructorArgs([
+				$this->mapper
+			])
+			->getMock();
+		$this->imageHooks = new ImageHooks(
+			$this->root,
+			$this->mapper,
+			$this->extractor,
+			$this->dbManager,
+			$this->container->query('ServerContainer')->getConfig()->getSystemValue('datadirectory')
+		);
+	}
+
+	public function testPostCreate() {
+		$location = '/testFolder/test.jpg';
+		$fileId = 260495;
+		$jpgFile = $this->mockJpgFile($location, $fileId);
+
+		$absolutePath = $this->container->query('ServerContainer')->getConfig()->getSystemValue('datadirectory').$location;
+
+		$this->assertEquals($jpgFile->getPath(), $location);
+
+		$metadata = null;
```

Comment:
Could you share some code?
I've just added an array and the test still passed.

Normally if you add a condition which returns or throws an exception in case of bad data, then you would be able to replicate it using the mocked extractor